File: APIWebVE/app/src/main/java/com/example/apiwebve/MainActivity.java
package com.example.apiwebve;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    Animation zoom;
    ImageView img;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        zoom = AnimationUtils.loadAnimation(getApplicationContext(), R.anim.zoom);
        img = findViewById(R.id.image);
        img.startAnimation(zoom);

        Handler handler = new Handler();

        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(getApplicationContext(),MainActivity2.class);
                startActivity(intent);
                finish();
            }
        }, 4000);
    }
}

File: RemoveBackgroundByAristiDevs/app/src/main/java/com/example/removebackgroundbyaristidevs/MainActivity.kt
package com.example.removebackgroundbyaristidevs

import android.content.ActivityNotFoundException
import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.MediaStore
import android.widget.ImageView
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.graphics.drawable.toBitmap
import com.example.removebackgroundbyaristidevs.databinding.ActivityMainBinding
import com.slowmac.autobackgroundremover.BackgroundRemover
import com.slowmac.autobackgroundremover.OnBackgroundChangeListener
import java.lang.Exception

/**
 * 1-Arquitectura
 * 2-Buscar diseño
 * 3-Poder seleccionar desde la galaria o desde la camara
 * 4-Guardar el resultado
 * 5-Opción de TRIM
 * 6-Compartir imagen
 * 7-Meter métodos de monetización (gratis con publi/pago único)
 * 8-Ficha
 * https://www.twitch.tv/videos/1647931043
 *45.40min
 * */
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    lateinit var iv: ImageView

    private val imageResult =
        registerForActivityResult(
            ActivityResultContracts.GetContent()
        ) { uri: Uri? ->
            uri?.let { uri ->
//                binding.img.setImageURI(uri)
                iv.setImageURI(uri)
                processImage()
            }

        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        binding = ActivityMainBinding.inflate(layoutInflater)
//        setContentView(binding.root)
        setContentView(R.layout.activity_main)
        iv = findViewById(R.id.ivTests)
        imageResult.launch("image/*")
    }

//    val REQUEST_IMAGE_CAPTURE = 1
//
//    private fun dispatchTakePictureIntent() {
//        val takePictureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
//        try {
//            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE)
//        } catch (e: ActivityNotFoundException) {
//            // display error state to the user
//        }
//    }

    fun processImage(/*image: Bitmap*/) {
        BackgroundRemover.bitmapForProcessing(
//            image,
            iv.drawable.toBitmap(),
            false,
            object : OnBackgroundChangeListener {
                override fun onSuccess(bitmap: Bitmap) {
                    // do what ever you want to do with this bitmap
                    iv.setImageBitmap(bitmap)
                }

                override fun onFailed(exception: Exception) {
                    //exception
                }
            }
        )
    }
}